### DISCUSSION TITLE
Poetry is not updating fastapi

### DISCUSSION THREAD

**Discussion by Popkornium18** [python-poetry/poetry#8579] [Q&A] [answered]
I have a project that uses fastapi and when I added it as a dependency 0.103.0 was the newest version. That means I have `fastapi = "^0.103.0"` as a dependency. I think that means that running `poetry update` should update it to the newest version, _unless_ that version is 1.0.0. Is that correct?

When I run `poetry update` it only updates fastapi to `0.103.2`. I have to bump the version in the pyproject.toml file, in order to update to the newest minor version.

Am I just assuming poetry works in ways it just doesn't or is there something wrong?

**Comment by dimbleby** [comment, marked as answer]
please read the docs https://python-poetry.org/docs/dependency-specification#caret-requirements

**Reply by Popkornium18** [reply]
Ah I see my mistake. I read that page and thought my use matched the first example. I didn't realize that the caret has a different meaning for versions lower than 1.0.
